### PR TITLE
[INV-95] Add wishlist functionality for Accommodation Card

### PR DESCRIPTION
**Pull request by ShahobiddinIsmoilov**
# Pull request for: [INV-95]

## Description

This pull request updates the response coming from list accommodations endpoint.

### Main changes:
* Created ```OptionalAccessTokenGuard``` for endpoints that should behave differently based on request authentication state.
* Added boolean ```isSavedToWishlist``` property to the ```Accommodation``` objects being returned.

### Demonstration
image

## Checklist

- [x] I have manually tested the changes and related features
- [x] I have run the linter and fixed any possible issues
- [x] I have run the tests and fixed any possible issues
- [x] I have run the build and fixed any possible issues

### CODE REVIEW

File: src/modules/accommodations/services/accommodations.service.ts [hunk 1]
```diff
@@ -49,14 +50,18 @@ export class AccommodationService {
...
-  async listAccommodations(filters: AccommodationFiltersReqQueryDto) {
+  async listAccommodations(filters: AccommodationFiltersReqQueryDto, accountId: string | null) {
     const queryBuilder = this.accommodationRepository
       .createQueryBuilder('accommodation')
       .leftJoinAndSelect('accommodation.address', 'address')
       .leftJoinAndSelect('accommodation.amenity', 'amenity')
       .leftJoinAndSelect('accommodation.images', 'image')
       .where('accommodation.deletedAt is NULL');
 
+    if (accountId) {
```
Comment (NegatiV-64) [inline src/modules/accommodations/services/accommodations.service.ts:61]: Why it's needed?

File: src/modules/accommodations/services/accommodations.service.ts [hunk 2]
```diff
@@ -49,14 +50,18 @@ export class AccommodationService {
...
+      queryBuilder.leftJoinAndSelect('accommodation.wishlist', 'wishlist', 'wishlist.accountId = :accountId', { accountId });
```
Comment (NegatiV-64) [inline src/modules/accommodations/services/accommodations.service.ts:62]: Better to offload the functionality to separate method/service

File: src/modules/auth/auth.module.ts [hunk 3]
```diff
@@ -18,6 +19,6 @@ import { RefreshTokenStrategy } from './strategies/refresh-token.strategy';
     PassportModule.register({ defaultStrategy: 'google' }),
   ],
   controllers: [AuthController],
-  providers: [AuthService, GoogleStrategy, AccessTokenStrategy, RefreshTokenStrategy],
+  providers: [AuthService, GoogleStrategy, AccessTokenStrategy, RefreshTokenStrategy, OptionalAccessTokenStrategy],
```
Comment (NegatiV-64) [inline src/modules/auth/auth.module.ts:22]: Why?

File: src/modules/auth/strategies/optional-access-token.strategy.ts
Comment (NegatiV-64) [inline src/modules/auth/strategies/optional-access-token.strategy.ts:1]: I don't think that we need this endpoint

File: src/shared/decorators/get-optional-account.decorator.ts
Comment (NegatiV-64) [inline src/shared/decorators/get-optional-account.decorator.ts:1]: I don't think that we need this guard

File: src/shared/configs/jwt-auth.config.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 export const JwtAuthConfig = {
   AccessTokenKey: 'access-token',
   RefreshTokenKey: 'refresh-token',
+  OptionalAccessTokenKey: 'optional-access-token',
```
Comment (NegatiV-64) [inline src/shared/configs/jwt-auth.config.ts:4]: No need for this

File: src/shared/guards/optional-access-token.guard.ts
Comment (NegatiV-64) [inline src/shared/guards/optional-access-token.guard.ts:1]: Don't think that we need it

**Review by NegatiV-64 — CHANGES_REQUESTED** [review]
Don't mix the endpoints and services. Better to have duplicate code that spaghetti code with 100500 if/else or ambiguity

File: src/modules/auth/auth.module.ts (hunk 3 above)
Comment (ShahobiddinIsmoilov) [reply src/modules/auth/auth.module.ts]: > Why?

Removed

File: src/modules/auth/strategies/optional-access-token.strategy.ts
Comment (ShahobiddinIsmoilov) [reply src/modules/auth/strategies/optional-access-token.strategy.ts:1]: > I don't think that we need this endpoint

Removed

File: src/shared/configs/jwt-auth.config.ts (hunk 4 above)
Comment (ShahobiddinIsmoilov) [reply src/shared/configs/jwt-auth.config.ts]: > No need for this

Removed

**Review by NegatiV-64 — APPROVED** [review]

**Pull request merged**